### PR TITLE
Update food

### PR DESCRIPTION
**Pull request by ryanmillergm**
**Adds update food test and method:**
 - Adds router patch method to foods.js
 - Adds update test
 - Adds end points and responses to readme
 
**The following checks have been completed:**
 - [ x] Tested new feature(s) as well as any feasible edge cases
 - [ x] Updated README for changes (new endpoints, new packages, etc)
 - [ x] Merged in the latest master & resolved all merge conflicts
 - [x ] Ran database migrations
 - [x ] Ran the test suite - all tests are passing (or maybe skipped)
 - [x ] Checked affected endpoints in Postman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,7 +63,84 @@ Body:
...
+Request:
+```
+POST /api/v1/foods
+Accept: application/json
+```
+Both name and calories are required fields.
+Pass in the following parameters:
+```
+{
+    "food": {
```
Comment (chakeresa) [inline README.md]: I added an issue to resolve this (we can do it on a separate PR) -- we're not nesting within a `food` key right now.

File: README.md [hunk 2]
```diff
@@ -63,7 +63,87 @@ Body:
...
+### Update an existing food by id in the database
+Request:
+```
+PATCH /api/v1/foods/:id
+Accept: application/json
+```
+Pass in the following parameters in the body of the request:
+```
+{
+    "food":
```
Comment (chakeresa) [inline README.md]: Added issue for this -- not actually reading in data this way right now. Can resolve on a different branch.

File: routes/api/v1/foods.js [hunk 3]
```diff
@@ -77,4 +77,25 @@ router.delete("/:id", function (req, res, next) {
   })
 });
 
+/*UPDATE a food given the id*/
+router.patch("/:id", function (req, res, next) {
+  return Food.update({
+    name: req.body.name,
```
Comment (chakeresa) [inline routes/api/v1/foods.js]: Should we be using Joi here too?

File: test/requests/foods/update.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+var app = require('../../../app');
+var request = require("supertest");
+var expect = require('chai').expect;
+var Food = require('../../../models').Food;
+
+describe('api v1 foods UPDATE', function () {
```
Comment (chakeresa) [inline test/requests/foods/update.test.js]:
```suggestion
describe('api v1 foods :id PATCH', function () {
```

File: test/requests/foods/update.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      .then(() => {
+        return request(app)
+          .patch(`/api/v1/foods/1`)
+          .send({
+            name: "frog legs",
+            calories: 250
+          })
+      })
+      .then(response => {
+        let updatedFood = response.body[1][0];
```
Comment (chakeresa) [inline test/requests/foods/update.test.js]: Why is it [1][0] vs just [0]?

File: test/requests/foods/update.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    it('returns 404 if no food has that id', (done) => {
+      request(app)
+      .patch('/api/v1/foods/54')
+      .then(response => {
+        expect(response.statusCode).to.equal(404);
+    
+        done();
+      })
+      done();
```
Comment (chakeresa) [inline test/requests/foods/update.test.js]:
```suggestion
```

File: README.md (hunk 2 above)
Comment (chakeresa) [reply README.md]: Resolved in this branch

File: routes/api/v1/foods.js (hunk 3 above)
Comment (chakeresa) [reply routes/api/v1/foods.js]: Let's add a test for only updating one attribute to make sure it acts as expected

File: test/requests/foods/update.test.js (hunk 5 above)
Comment (chakeresa) [reply test/requests/foods/update.test.js]: Resolved on branch

File: routes/api/v1/foods.js (hunk 3 above)
Comment (chakeresa) [reply routes/api/v1/foods.js]: Added test on branch

**Comment by chakeresa** [comment]
For documentation, this PR resolves #4

**Comment by chakeresa** [comment]
This PR also resolves #25
